azavea/raster-foundry
# Feature/cmb/add scenes api

This commit adds API endpoints for creating, reading, updating, and
deleting scenes.

It is possible to sort scenes using any number of the following
attributes (assuming that the attributes exist):
 - creation time
 - modification time
 - organization
 - datasource
 - month of acquisition
 - acquisition time
 - sun azimuth
 - sun elevation
 - cloud cover

Scenes can be filtered based on:
 - creation time
 - modification time
 - acquisition month
 - cloud cover
 - sun azumuth
 - sun elevation
 - organization
 - datasource

### Testing

- Run migrations, start up server
- Obtain a valid JWT to validate requests
- Create a few scenes using a rest console or swagger (you can use the following JSON as a template)
 ```
{
  "thumbnailStatus": "PROCESSING",
  "tags": ["Test"],
  "ingestSizeBytes": 0,
  "acquisitionDate": "2016-09-15T12:49:41.324Z",
  "cloudCover": 10.800000190734863,
  "resolutionMeters": 20.0,
  "boundaryStatus": "PROCESSING",
  "sceneMetadata": {
    "test": 1
  },
  "status": "PROCESSING",
  "organizationId": "dfac6307-b5ef-43f7-beda-b9f208bb7726",
  "modifiedBy": "default",
  "visibility": "PUBLIC",
  "datasource": "NASA"
}
```
- Query endpoint with various filters and sorting applied
- Update one of the created scenes and verify that update is applied
- Also run tests

Additional tests can be written once #459 is fixed -- without the ability to insert scenes into the database outside of the API testing is a little difficult for the detail views.
